public/app/themes/clarity/inc/pagination-newscategory.php

```diff
@@ -6,41 +6,47 @@ function get_newscategory_pagination($category_id)
     $oAgency = new Agency();
     $activeAgency = $oAgency->getCurrentAgency();
 
-    /*
-    * A temporary measure so that API calls do not get blocked by
-    * changing IPs not whitelisted. All calls are within container.
-    */
-    $siteurl = 'http://127.0.0.1';
-
-    $post_per_page      = 'per_page=10';
-    $current_page       = '&page=1';
-    $agency_name        = '&agency=' . $activeAgency['wp_tag_id'];
-    $category_name      = '&news_category=' .$category_id;
-
-    $response = wp_remote_get($siteurl.'/wp-json/wp/v2/news/?' . $post_per_page . $current_page . $agency_name . $category_name);
-
-    if (is_wp_error($response)) {
-        return;
-    }
-
-    $pagetotal = wp_remote_retrieve_header($response, 'x-wp-totalpages'); ?>
-
-        <div id="load_more"></div>
-        <nav class="c-pagination category" role="navigation" aria-label="Pagination Navigation">
-        <?php if ($pagetotal > 0) {
-            ?>
-            <button class="more-btn" data-page="1" data-date="">
-            <span class="c-pagination__main "><span class="u-icon u-icon--circle-down"></span> Load Next 10 Results</span><span class="c-pagination__count"> 1 of <?php echo $pagetotal; ?></span>
-            </button>
-
-            <?php
-        } else {
-            ?>
+    $post_per_page = 10;
+
+    $args = [
+        'numberposts' => $post_per_page,
+        'post_type' => 'news',
+        'post_status' => 'publish',
+        'tax_query' => [
+          'relation' => 'AND',
+          [
+            'taxonomy' => 'agency',
+            'field' => 'term_id',
+            'terms' => $activeAgency['wp_tag_id']
+          ],
+          ...( $category_id ? [
+            'taxonomy' => 'news_category',
+            'field' => 'category_id',
+            'terms' =>  $category_id,
+          ] : []),
+      ]
+    ];
+
+    $query = new WP_QUERY($args);
```

Comment:
Can we check this?
The call follows the constant convention, as in all uppercase, yet, it is acting like a class.

The class I think you need is `WP_Query()` - which in turn might have some drawbacks.

The class resets the post object when called. which means generally, we need to reset data after getting our results... this may not matter here but wanted to flag that just in case.

```php
<?php
   $temp_post = $post; // Storing the object temporarily
   $my_query = new WP_Query();
   while($my_query->have_posts()) {
      // Loop in here
   }
   $post = $temp_post; // Restore the value of $post to the original
?>
```

Reply:
Thanks @wilson1000, I've updated this and added `wp_reset_postdata();` where I've used it as recommended in the docs, but let me know if I've missed something.